src/pool/srv_pool.c

```diff
@@ -925,14 +928,52 @@ handle_event(struct pool_svc *svc, struct pool_svc_event *event)
 		goto out_prop;
 	}
 
-	rc = pool_svc_exclude_rank(svc, event->psv_rank);
-	if (rc != 0) {
-		D_ERROR(DF_UUID": failed to exclude rank %u: "DF_RC"\n", DP_UUID(svc->ps_uuid),
-			event->psv_rank, DP_RC(rc));
-		goto out_prop;
-	}
+	if (DAOS_POOL_SUPPORT_REINT_ONLINE == 0)
+		goto DEAD;	
 
-	D_DEBUG(DB_MD, DF_UUID": excluded rank %u\n", DP_UUID(svc->ps_uuid), event->psv_rank);
+	if (event->psv_type == CRT_EVT_ALIVE) {
+        dom = pool_map_find_node_by_rank(svc->ps_pool->sp_map, event->psv_rank);
+		if (dom == NULL) {
+			D_ERROR("rank %u not in map \n",event->psv_rank);
+			goto out_prop;
+		}
+		
+        if (dom->do_comp.co_status != PO_COMP_ST_DOWN && dom->do_comp.co_status != PO_COMP_ST_DOWNOUT) {
+			D_DEBUG(DB_MD, "rank %u status is ok,don't need up!\n",event->psv_rank);
+            goto out_prop;
+        }
+	
+		rc = pool_svc_reint_rank(svc, event->psv_rank);
+        if (rc != 0) {
+			D_ERROR(DF_UUID": failed to reint rank %u: "DF_RC"\n", DP_UUID(svc->ps_uuid),
+			    event->psv_rank, DP_RC(rc));
+		    goto out_prop;
+        }
+	
+		 D_DEBUG(DB_MD, DF_UUID": reint rank %u\n", DP_UUID(svc->ps_uuid), event->psv_rank);
+	}
+
+DEAD:
```

Comment:
[Nit] Could we use the following structure instead, please?

    ABT_rwlock_rdlock(sp_lock);
    node = pool_map_find_node_by_rank(...);
    status = node->do_comp.co_status;
    ABT_rwlock_unlock(sp_lock);
    
    if (event->psv_type == DEAD) {
      if (status == ... || ...)
        goto out_prop;
      rc = pool_svc_exclude_rank(...);
      ...
    } else { /* ALIVE */
      if (status != ... || ...)
        goto out_prop;
      rc = pool_svc_reint_rank(...);
      ...
    }